### DISCUSSION TITLE
In a commit that is part of a PR from outside repository, #xx references my issue, not theirs?

### DISCUSSION THREAD

**Discussion by mrks** [community/community#22267] [New to GitHub]
<p>Our organization’s repo has a PR from a forked repo. In that PR, commits like “yaddayadda, fixes <span class="hashtag">#123</span>” are pushed. Github now references our issue 123 in the commit, even though the committer meant to close an issue created in their fork. Who needs to fix this? Is this something that I can prevent as a repo owner accepting PRs or is this something on their end? If it makes any difference, their repo is not publicly visible, while ours is.</p>

**Comment by lee-dohm** [comment]
<p>In forks Issues are disabled by default because the idea is that people working in forks are generally working off of the parent repo’s issue list. So in the normal case, the <span class="hashtag">#xx</span> just works. If they manually enabled Issues on their fork, track them separately from the parent repository, but still merge back to the parent, they might want to use a fully-qualified issue link like <code>foo/bar#123</code> or just the full URL such as <code>[https://github.com/foo/bar/issues/123](https://github.com/foo/bar/issues/123)</code>. The closing magic will still work, even though their repo is not publicly visible.</p>
<p>As far as what you can do as a repo owner when someone puts an invalid “fixes <span class="hashtag">#xxx</span>” in their commit message, you have a few options:</p>
<ol>
<li>If they have left “Allow edits from maintainers” enabled (it is enabled by default), then you can interactively rebase the branch, edit the commit message, and force-push the change</li>
<li>You can request that they do the work of changing their commit message and update the PR</li>
<li>You can merge the PR and manually reopen the referenced issue</li>
</ol>
<p>If they put it in the body of the PR, then you can always edit it out before merging the PR.</p>
<p>I hope that helps! Let us know if you have any other questions.</p>

**Comment by mrks** [comment]
<p>Thanks for your answer. I think I now understand what happens. Is there any way we could get a warning in a PR when the origin and target repos both have issues enabled and the <code>#123</code> might reference different issues depending on context?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>I’ll pass along the feedback. I can’t promise when or even if it will be implemented, but I’ll let the appropriate team know.</p>
